instance/gandi.py

```diff
@@ -189,4 +192,157 @@ def remove_dns_record_callback(zone_id, zone_version):
         )
 
 
-api = GandiAPI()
+class GandiV5API:
+    """
+    Gandi v5 LiveDNS API client
+    """
+    def __init__(self, api_key, api_base='https://dns.api.gandi.net/api/v5'):
+        self.api_base = api_base
+        self.api_key = api_key
+        self._domain_cache = None
```

Comment:
Shouldn't we populate the domain cache when instancing the class? 
Can you add a call to `_populate_domain_cache` here?

Reply:
@giovannicimolin, the domain cache is needed only to split the given hostname into the registered domain name and its subdomain in the `_split_domain_name()` method and that's why it is called there before performing the split operation.